src/libstd/thread/mod.rs

```diff
@@ -386,6 +386,14 @@ impl Builder {
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn spawn<F, T>(self, f: F) -> io::Result<JoinHandle<T>> where
         F: FnOnce() -> T, F: Send + 'static, T: Send + 'static
+    {
+        unsafe { self.spawn_unchecked(f) }
+    }
+
+    /// FIXME: Doc
+    #[unstable(feature = "thread_spawn_unchecked", issue = "0")]
+    pub unsafe fn spawn_unchecked<'a, F, T>(self, f: F) -> io::Result<JoinHandle<T>> where
+        F: FnOnce() -> T, F: Send + 'a, T: Send + 'a
```

Comment:
Why the `+'a` bounds? AFAIK these are useless. No bound means no restriction.

(For trait objects there are default lifetime bounds being added, but that's because they have to -- for where clauses, no such thing happens.)

Reply:
I believe you are right, these lifetime bounds can probably be elided/implicitly inferred by the compiler, will check & fix.